src/svg.rs

```diff
@@ -33,13 +35,39 @@ impl SvgVersion {
 declare_surface!(SvgSurface, SurfaceType::Svg);
 
 impl SvgSurface {
-    pub fn new<P: AsRef<Path>>(width: f64, height: f64, path: P) -> Result<SvgSurface, Status> {
-        let path = path.as_ref().to_string_lossy().into_owned();
-        let path = CString::new(path).unwrap();
+    pub fn new<P: AsRef<Path>>(
+        width: f64,
+        height: f64,
+        path: Option<P>,
+    ) -> Result<SvgSurface, Status> {
+        #[cfg(not(windows))]
+        let path = {
+            let path = path.map(|p| {
+                CString::new(p.as_ref().as_os_str().as_bytes())
+                    .expect("Invalid path with NULL bytes")
+            });
+            path.map(|p| p.as_ptr()).unwrap_or(ptr::null())
+        };
+        #[cfg(windows)]
+        let path = {
+            // Based on GLib path to C
+            let path_str = path.map(|p| {
+                p.as_ref()
+                    .to_str()
+                    .expect("Path can't be represented as UTF-8")
+                    .to_owned()
+            });
+            if path_str.starts_with("\\\\?\\") {
```

Comment:
This and the following lines need to stay inside the `map` closure. You still have an `Option` here. Later you need to also get the pointer of the whole thing